src/hooks/routing/useRouterTrade.ts

```diff
@@ -40,94 +39,60 @@ export function useRouterTrade<TTradeType extends TradeType>(
   gasUseEstimateUSD?: CurrencyAmount<Token>
 } {
   const { provider } = useWeb3React()
-  const isWindowVisible = useIsWindowVisible()
-  // Debounce is used to prevent excessive requests to SOR, as it is data intensive.
-  // Fast user actions (ie updating the input) should be debounced, but currency changes should not.
-  const [debouncedAmount, debouncedOtherCurrency] = useDebounce(
-    useMemo(() => [amountSpecified, otherCurrency], [amountSpecified, otherCurrency]),
-    200
-  )
-  const isDebouncing = amountSpecified !== debouncedAmount && otherCurrency === debouncedOtherCurrency
-  const debouncedAmountSpecified = isWindowVisible ? debouncedAmount : undefined
-
-  const [currencyIn, currencyOut]: [Currency | undefined, Currency | undefined] = useMemo(
-    () =>
-      isExactInput(tradeType)
-        ? [debouncedAmountSpecified?.currency, debouncedOtherCurrency]
-        : [debouncedOtherCurrency, debouncedAmountSpecified?.currency],
-    [debouncedAmountSpecified, debouncedOtherCurrency, tradeType]
-  )
-
+  const [currencyIn, currencyOut] = isExactInput(tradeType)
+    ? [amountSpecified?.currency, otherCurrency]
+    : [otherCurrency, amountSpecified?.currency]
   const queryArgs = useRouterArguments({
     tokenIn: currencyIn,
     tokenOut: currencyOut,
-    amount: debouncedAmountSpecified,
+    amount: amountSpecified,
     tradeType,
     routerUrl,
     provider: provider as JsonRpcProvider,
   })
 
-  const { isError, data, currentData } = useGetQuoteQuery(queryArgs ?? skipToken, {
-    // Price-fetching is informational and costly, so it's done less frequently.
+  // Get the cached state *immediately* to update the UI without sending a request - using useGetQuoteQueryState -
+  // but debounce the actual request - using useLazyGetQuoteQuery - to avoid flooding the router / JSON-RPC endpoints.
+  const { isError, data, currentData } = useGetQuoteQueryState(queryArgs)
+  const isValidBlock = useIsValidBlock(Number(data?.blockNumber))
+  const isSyncing = currentData !== data || !isValidBlock
+
+  const [trigger] = useLazyGetQuoteQuery({
+    // PRICE fetching is informational and costly, so it's done less frequently.
     pollingInterval: routerPreference === RouterPreference.PRICE ? ms`2m` : ms`15s`,
```

Comment:
how does polling work with lazy queries? Does it start polling on the first `trigger` call?

Reply:
> Accepts polling/re-fetching options to trigger automatic re-fetches when the corresponding criteria is met and the fetch has been manually called at least once

https://redux-toolkit.js.org/rtk-query/api/created-api/hooks#uselazyquery

Reply:
👍 bit of a confusing behavior but it was my intuition